### PR TITLE
Add RHEL7 compatibility for gzip options

### LINKED ISSUE #11047: Build fails due to invalid gzip option -k
### What did you do?

Attempt to build Prometheus on a CentOS 7 platform

### What did you expect to see?

A successful build. It last worked on version 2.30.3 that I've tested.

### What did you see instead? Under which circumstances?

```
gzip: invalid option -- 'k'
Try `gzip --help' for more information.
>> building binaries
/home/kitchen/recipes/prometheus/tmp-build/bin/promu build --prefix /home/kitchen/recipes/prometheus/tmp-build/src/github.com/prometheus/prometheus
 >   prometheus
# github.com/prometheus/prometheus/web/ui
web/ui/embed.go:21:3: usage: //go:embed pattern...
!! command failed: build -o /home/kitchen/recipes/prometheus/tmp-build/src/github.com/prometheus/prometheus/prometheus -ldflags -X github.com/prometheus/common/version.Version=2.37.0 -X github.com/prometheus/common/version.Revision=b41e0750abf5cc18d8233161560731de05199330 -X github.com/prometheus/common/version.Branch=HEAD -X github.com/prometheus/common/version.BuildUser=kitchen@3460a3706333 -X github.com/prometheus/common/version.BuildDate=20220721-14:47:40  -extldflags '-static' -a -tags netgo,builtinassets github.com/prometheus/prometheus/cmd/prometheus: exit status 2
make: *** [common-build] Error 1
```

This comes from:
```
scripts/compress_assets.sh:find static -type f -exec gzip -fk '{}' \; -print0 | xargs -0 -I % echo %.gz | xargs echo //go:embed >> embed.go
```

It seems when building on CentOS 7, the available gzip version does not include the `-k` option, which causes the build to fail. Builds should not be host dependent, outside of the Go / Node versions.

### System information

CentOS 7

### Prometheus version

```text
2.37.0
```

### Prometheus configuration file

```yaml
N/A
```

### Alertmanager version

```text
N/A
```

### Alertmanager configuration file

```yaml
N/A
```

### Logs

```text
N/A
```

### PR DESCRIPTION
**Pull request by mttradebyte**
Fixes #11047

### CODE REVIEW

**Comment by mttradebyte** [comment]
Are you suggesting checking the return value of `gzip -k -h`? That will return `1` but gzip might also return `1` for other issues, so surely it would be less accurate?

**Comment by roidelapluie** [comment]
Just running `gzip -h -k` (no extra args) returns 0 when `-k` is supported.

**Comment by mttradebyte** [comment]
Updated and tested on Pop_OS 22.04, RHEL7 and CentOS 7 - all produce the expected results.

File: scripts/compress_assets.sh [hunk 1]
```diff
@@ -6,6 +6,10 @@ set -euo pipefail
 
 cd web/ui
 cp embed.go.tmpl embed.go
+
+# gzip option '-k' may not always exist in the latest gzip available on different distros
```
Comment (roidelapluie) [inline scripts/compress_assets.sh:10]:
```suggestion
# gzip option '-k' may not always exist in the latest gzip available on different distros.
```

File: scripts/compress_assets.sh [hunk 2]
```diff
@@ -6,6 +6,10 @@ set -euo pipefail
...
+GZIP_OPTS="-fk" && [[ ! $(gzip -k -h 2>/dev/null) ]] && GZIP_OPTS="-f"
```
Comment (roidelapluie) [inline scripts/compress_assets.sh:11]: Can we expand that for readability?
```suggestion
GZIP_OPTS="-fk"
# gzip option '-k' may not always exist in the latest gzip available on different distros.
if ! gzip -k -h &>/dev/null; then GZIP_OPTS="-f"; fi
```

**Comment by roidelapluie** [comment]
It seems we have the comment twice, can you remove one? then I can merge

**Comment by mttradebyte** [comment]
It's the same comment but refers to different pieces of code as I did it from within Github. I'm not used to the whole PR stuff in Github as I don't use it day to day, so what do I need to do exactly?

File: scripts/compress_assets.sh [hunk 3]
```diff
@@ -6,6 +6,12 @@ set -euo pipefail
 
 cd web/ui
 cp embed.go.tmpl embed.go
+
+# gzip option '-k' may not always exist in the latest gzip available on different distros.
```
Comment (roidelapluie) [inline scripts/compress_assets.sh:10]:
```suggestion
```

**Comment by mttradebyte** [comment]
Sorry, been jumping between so many things since getting back from holiday, I didn't realise you literally meant the comment in the code, not the commit message :laughing:. That's done now.

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**